src/plugins/wmr/plugin.js

```diff
@@ -65,33 +66,41 @@ export default function wmrPlugin({ hot = true } = {}) {
 				before += `const module={${hot ? 'hot:import.meta.hot' : ''}};\n`;
 			}
 
+			const hasExport = code.match(/\bexport\b/);
+
 			// Detect modules that appear to have both JSX and an export, and inject prefresh:
 			// @todo: move to separate plugin.
 			// if (code.match(/\/\*@@prefresh_include\*\//) && code.match(/\bexport\b/)) {
-			if (code.match(/html`[^`]*<([a-zA-Z][a-zA-Z0-9.:-]*|\$\{.+?\})[^>]*>/) && code.match(/\bexport\b/)) {
+			if (code.match(/html`[^`]*<([a-zA-Z][a-zA-Z0-9.:-]*|\$\{.+?\})[^>]*>/) && hasExport) {
 				// if (this.getModuleInfo(id).hasJSX) {
 				hasHot = true;
 				after += '\n' + PREFRESH;
 				// }
 			}
 
-			if (!hasHot) return null;
+			if ((!hot && !hasHot) || (!hasExport && !code.match(/\bimport\b/))) return null;
```

Comment:
`hasHot` can be true in prod, but we will already have injected `null` for the HMR API on line 52, so we should still bail out here.

```suggestion
			if (!hot || !hasExport) return null;
```